### PR TITLE
Clean up game engine actions and reducers

### PR DESCRIPTION
**Pull request by siliconcupcake**

### CODE REVIEW

File: src/game-engine/eventListener.js [hunk 1]
```diff
@@ -20,37 +20,32 @@ var startCoreGameEventListeners = (dispatch) => {
...
 		}
 	})
 	socket.on('game-updates', (response) => {
+		console.log(response)
 		switch (response.type) {
 			case 'CREATE':
 				if (response.code === 200) dispatch(createGameSuccess(response))
 				else dispatch(createGameFailure(response))
 				break
-			case 'LIST':
```
Comment (NavenAllen) [inline src/game-engine/eventListener.js:35]: You get the players data in LIST right? JOIN is the action that is sent to all players to indicate that a player has joined.

File: src/game-engine/state/actions/game.actions.js [hunk 2]
```diff
@@ -132,29 +125,28 @@ function joinGameRequest(user) {
...
 }
 
 function joinGameFailure(data) {
-	localStorage.removeItem('gameCode')
 	return (dispatch) =>
 		dispatch({ type: gameConstants.JOIN_GAME_FAILURE, data })
 }
 
 function leaveGameRequest(code, pid) {
 	return (dispatch) => {
```
Comment (NavenAllen) [inline src/game-engine/state/actions/game.actions.js:146]: Shouldn't we also remove gameCode from localStorage when a person leaves?

File: src/game-engine/state/reducers/game.reducer.js [hunk 3]
```diff
@@ -105,6 +96,11 @@ export function game(state = initialState, action) {
 				inGame: false
 			}
 		case gameConstants.LEAVE_GAME_REQUEST:
```
Comment (NavenAllen) [inline src/game-engine/state/reducers/game.reducer.js:98]: You removed LEAVE_GAME_REQUEST dispatch from the action, so why have a reducer here? Why was it removed, tho? o.O

File: src/game-engine/state/actions/game.actions.js [hunk 4]
```diff
@@ -190,29 +182,14 @@ function startGameFailure(data) {
...
+		localStorage.removeItem('gameCode')
+		socket.emit('destroy', {
 			code,
 			pid
 		})
-		dispatch({ type: gameConstants.DESTROY_GAME_REQUEST })
 	}
 }
 
-function destroyGameSuccess(data) {
```
Comment (NavenAllen) [inline src/game-engine/state/actions/game.actions.js:201]: Why did you remove the destroy game actions?

File: src/game-engine/state/reducers/game.reducer.js [hunk 5]
```diff
@@ -105,6 +96,11 @@ export function game(state = initialState, action) {
...
+			return {
+				...state,
+				locked: true
+			}
+		case gameConstants.LEAVE_GAME_SUCCESS:
```
Comment (NavenAllen) [inline src/game-engine/state/reducers/game.reducer.js:103]: Shouldn't lock be false if leave game is success? And we should also remove all the store data right

**Review by NavenAllen — CHANGES_REQUESTED** [review]

File: src/game-engine/eventListener.js (hunk 1 above)
Comment (siliconcupcake) [reply src/game-engine/eventListener.js:35]: Ohh, I made corresponding changes in backend also, will push that soon.

File: src/game-engine/state/actions/game.actions.js (hunk 2 above)
Comment (siliconcupcake) [reply src/game-engine/state/actions/game.actions.js:146]: Yes, will add.

File: src/game-engine/state/reducers/game.reducer.js (hunk 3 above)
Comment (siliconcupcake) [reply src/game-engine/state/reducers/game.reducer.js:98]: I only removed Destroy Game Request right ?
Leave Game is still there.

File: src/game-engine/state/actions/game.actions.js (hunk 4 above)
Comment (siliconcupcake) [reply src/game-engine/state/actions/game.actions.js:201]: Destroy game won't return a response from backend. You can just send a request and forget all about it. If it doesn't destroy in backend there's nothing you can do about it from frontend anyways.

File: src/game-engine/state/reducers/game.reducer.js (hunk 3 above)
Comment (NavenAllen) [reply src/game-engine/state/reducers/game.reducer.js:98]: Oh yeah, sorry

**Review by NavenAllen — APPROVED** [review]
